docs/proposals/custom-metrics.md

```diff
@@ -0,0 +1,119 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/custom-metrics.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Custom Metrics in Kubernetes
+
+
+## Preface
+
+Our aim is to create a mechanism in Kubernetes that will allow pods to expose custom metrics, collect them, and make them accessible.
+Custom metrics are needed by:
+* horizontal pod autoscaler, for autoscaling the number of pods based on them;
+* scheduler, for using them in a sophisticated scheduling algorithm;
+* general monitoring of application health by a user (we are not sure yet whether this should be handled separately, by a different,
+general monitoring solution, see [#665](https://github.com/kubernetes/heapster/issues/665));
+* ...
+
+
+## Design
+
+For the Kubernetes version 1.2, we plan to implement aggregation of pod custom metrics in Prometheus format by pull.
+
+Each pod, to expose custom metrics, will expose a set of Prometheus endpoints.
+CAdvisor will collect metrics from such endpoints of all pods on each node by pulling, and expose them to Heapster.
```

Comment:
I think the decision was that the custom endpoint exposed by the pod would be exposed and accessible.

We need to add something to the docs to outline that these custom metrics are not private information.

Ideally, we would have something setup where the prometheus endpoint could only be accessible by the kubelet, but I don't believe this was ever planned for the first iteration.